pkg/apis/core/v1alpha1/dockerimage_types.go

```diff
@@ -186,6 +186,30 @@ func (in *DockerImageList) GetListMeta() *metav1.ListMeta {
 
 // DockerImageStatus defines the observed state of DockerImage
 type DockerImageStatus struct {
+	// A fully-qualified image reference of a built image, as seen from the local
+	// network.
+	//
+	// Usually includes a name and an immutable tag.
+	//
+	// NB: If we're building to a particular registry, this may
+	// have a different hostname from the Spec `Ref` field.
+	Ref string `json:"ref,omitempty" protobuf:"bytes,1,opt,name=ref"`
```

Comment:
should this should be `+optional`? IIUC, during the first build, we'll want to set `Building` but not `Ref`.

Reply:
^ That's interesting - should the `Ref` actually be part of `DockerImageStateCompleted`?

Reply:
I don't think so - I think when one build has finished and the next build is running, we want Ref pointing at the last successful build and the DockerImageStates set describing the current build. i.e., there would be no DockerImageStateCompleted (or it might have an error). It might be good to add some comments to clear this up.

Reply:
ya, I could go either way on this. We could also have it in both fields, similar to how ContainerStatus has some things duplicated. it doesn't really matter that much, since we expect to write the result to ImageMapStatus